From 73b8a457887f0a1e99fad46c60788d16ab95c76a Mon Sep 17 00:00:00 2001
From: Jing Zhao <jing9@apache.org>
Date: Wed, 2 Sep 2015 15:19:04 -0700
Subject: [PATCH 2647/2848] HDFS-9003. ForkJoin thread pool leaks. Contributed
 by Kihwal Lee.

(cherry picked from commit de928d566a119f0b7fa5f171719642cd86be0af7)
(cherry picked from commit 1d56325a80cdac5820079ac04fa18a7e5126f8ef)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt
(cherry picked from commit 879567d24a6e3b625f639a88137fb1654c54c18f)

Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt
(cherry picked from commit f638ff90420768d54bcbad0ac4a1850ce001611d)

 Conflicts:
	hadoop-hdfs-project/hadoop-hdfs/CHANGES.txt

Change-Id: I1c56d5e79e6529fbc541c9b2b90c3fd053ef933d
---
 .../hadoop/hdfs/server/namenode/FSImage.java       |    1 +
 1 file changed, 1 insertion(+)

diff --git a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java
index b3a4c54..9d3103e 100644
--- a/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java
+++ b/hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/namenode/FSImage.java
@@ -939,6 +939,7 @@ static void updateCountForQuota(INodeDirectory root, int threads) {
     RecursiveAction task = new InitQuotaTask(root, counts);
     p.execute(task);
     task.join();
+    p.shutdown();
     LOG.info("Quota initialization completed in " + (Time.now() - start) +
         " milliseconds\n" + counts);
   }
-- 
1.7.9.5

